syntax = "proto3";

package order;

import "google/protobuf/empty.proto";


service OrderService {
    rpc CreateOrder (OrderCreateRequest) returns (OrderResponse);
    rpc GetOrder (OrderId) returns (OrderResponse);
    rpc CancelOrder (OrderId) returns (google.protobuf.Empty);
}

enum Status {
    CREATED = 0;
    AWAITING_PAYMENT = 1;
    PAID = 2;
    COMPLETED = 3;
    CANCELLED = 4;
}

message OrderId {
    string id = 1;
}

message OrderCreateRequest {
    double amount = 1;
}

message OrderResponse {
    string id = 1;
    Status status = 2;
    double amount = 3;
}